### PR TITLE
Add methods to FrameAllocator for allocating with a given alignment.

### LINKED ISSUE #21: Allow alignment to be specified for `FrameAllocator`
For some usecases, e.g. allocating MMIO space for PCI BARs, it is necessary to allocate address space with both a particular size and a particular alignment. It would be useful if `FrameAllocator::alloc` took an alignment parameter to specify what alignment is required.

### PR DESCRIPTION
**Pull request by qwandor**
These take a Layout, for consistency with other allocators.

Fixes #21.

### CODE REVIEW

File: src/frame.rs [hunk 1]
```diff
@@ -67,14 +68,27 @@ impl FrameAllocator {
...
     }
 
-    /// Add a range of frame to the allocator
+    /// Add a range of frames to the allocator.
     pub fn insert(&mut self, range: Range<usize>) {
         self.add_frame(range.start, range.end);
     }
 
-    /// Alloc a range of frames from the allocator, return the first frame of the allocated range
+    /// Allocate a range of frames from the allocator, return the first frame of the allocated range
```
Comment (filnet) [inline src/frame.rs:76]: missing period :)

Comment (qwandor) [reply src/frame.rs]: Done.

**Review by jiegec — APPROVED** [review]

**Comment by jiegec** [comment]
Please resolve merge conflicts.

**Comment by qwandor** [comment]
> Please resolve merge conflicts.

Done.

**Pull request merged**